lib/Runtime/ByteCode/ByteCodeEmitter.cpp

```diff
@@ -3064,6 +3064,15 @@ void ByteCodeGenerator::EmitOneFunction(ParseNodeFnc *pnodeFnc)
             DefineFunctions(funcInfo);
         }
 
+        // insert a yield at the top of a module body so that function definitions can be hoisted accross modules
+        if (pnodeFnc->IsModule())
+        {
+            Js::RegSlot tempReg = funcInfo->AcquireTmpRegister();
```

Comment:
Oh one other idea. Instead of doing this here, we could add a synthetic yield ParseNode to the beginning of the body of the module function during parse. That might have a very similar effect to adding this bytecode here. You'd want to add the yield parse node after we parse the module body source in `GenerateModuleFunctionWrapper`. What do you think?

Reply:
This sounds good but I can't work out how to do it.

In particular I cannot work out how to add a yield node to the start of the function's parseNodeList after it has been parsed.

Reply:
For now I've put in a helper method in the ByteCodeEmitter - this has the slight benefit of skipping the NewScObject opCode that the normal EmitYield path includes.